sql/analyzer/resolve_columns.go

```diff
@@ -240,12 +240,14 @@ func dedupStrings(in []string) []string {
 }
 
 // findOnDupUpdateLeftExprs gathers all the left expressions for statements in InsertInto.OnDupExprs
-// the
-func findOnDupUpdateLeftExprs(onDupExprs []sql.Expression) map[sql.Expression]bool {
-	onDupUpdateLeftExprs := map[sql.Expression]bool{}
+// onDupExprs are always set with a column on the left
+func findOnDupUpdateLeftExprs(onDupExprs []sql.Expression) map[*expression.UnresolvedColumn]bool {
+	onDupUpdateLeftExprs := map[*expression.UnresolvedColumn]bool{}
```

Comment:
are we sure the set of acceptable expressions is only UnresolvedColumn? It can't be like, `1 + letter`? Or a bind var/user var?